bin/build_all_jenkins.sh

```diff
@@ -14,5 +14,6 @@ pip3 install numpy scikit-build
 # Configure and build source, continuing if there are errors
 rm -rf build
 mkdir build
-cmake -B build -S . -DCMAKE_EXPORT_COMPILE_COMMANDS=1 "$@"
-make -C build -k -j $(nproc)
+cd build
+cmake -DPYTHON_EXECUTABLE="$(which python3)" -DCMAKE_EXPORT_COMPILE_COMMANDS=1 "$@" ..
```

Comment:
the virtualenv should symlink ``python`` to whichever python executable you created the environment with so I don't see why this would be necessary unless cmake follows some very sketch python-finding logic

Reply:
I do remember in the previous cmake we used a python-based lookup to find numpy - _maybe_ that was because the cmake python-finding isn't virtualenv-aware (which would be kinda shit)

Reply:
It finds the python binary correctly on my machine when in a virtualenv but I suspect that it's still preferentially looking for python 2. I've told cmake we want python v3+ everywhere and that seemed to fix it, so let's just deprecate support for python 2.